src/medication-dispense/medication-dispense.resource.tsx

```diff
@@ -0,0 +1,85 @@
+import { fhirBaseUrl, openmrsFetch, Session } from "@openmrs/esm-framework";
+import dayjs from "dayjs";
+import useSWR from "swr";
+import { MedicationDispense, MedicationRequest, OrderConfig } from "../types";
+
+export function saveMedicationDispense(
+  medicationDispense: MedicationDispense,
+  abortController: AbortController
+) {
+  return openmrsFetch(`${fhirBaseUrl}/MedicationDispense`, {
+    method: "POST",
+    signal: abortController.signal,
+    headers: {
+      "Content-Type": "application/json",
+    },
+    body: medicationDispense,
+  });
+}
+
+export function useOrderConfig() {
+  const { data, error, isValidating } = useSWR<{ data: OrderConfig }, Error>(
+    `/ws/rest/v1/orderentryconfig`,
+    openmrsFetch
+  );
+  return {
+    orderConfigObject: data ? data.data : null,
+    isLoading: !data && !error,
+    isError: error,
+    isValidating,
+  };
+}
+
+export function initiateMedicationDispenseBody(
+  medicationRequests: Array<MedicationRequest>,
+  session: Session
+) {
+  let dispenseBody = [];
+  medicationRequests.map((medicationRequest) => {
+    let dispense = {
+      resourceType: "MedicationDispense",
+      status: "on-hold", // might need to change this to appropriate status
```

Comment:
yeah, I think this will need to be a coding reference (uuid or concept mapping) to an OpenMRS Concept in order to work... but, I don't think we defined exactly how we want this to work yet and I think it's fine to ticket this as a future enhancement.  Probably better to just not set the status at all for now, as I don't think "on-hold" will do anything/work?

Reply:
The documentation had this as a string along and it seems to be related to the ones defined in the list Tabs. We had only configured `on-hold` for testing purposes. This value is also related to the ongoing talk thread regarding the what statuses to support



> "[status](https://www.hl7.org/fhir/medicationdispense-definitions.html#MedicationDispense.status)" : "<[code](https://www.hl7.org/fhir/datatypes.html#code)>", // R!  [preparation | in-progress | cancelled | on-hold | completed | entered-in-error | stopped | declined | unknown](https://www.hl7.org/fhir/valueset-medicationdispense-status.html)